modules/apps/headless/headless-delivery/headless-delivery-test/src/testFunctional/tests/SiteDocumentEntry.testcase

```diff
@@ -240,4 +278,108 @@ definition {
 		}
 	}
 
+	@priority = 3
+	test CanNotUpdateDocumentWithInvalidFileName {
+		task ("Given a document in a site") {
+			var siteId = JSONGroupAPI._getGroupIdByNameNoSelenium(
+				groupName = "Guest",
+				site = "true");
+			var filePath = TestCase.getDependenciesDirPath(fileName = "Document_1.jpg");
+
+			var response = DocumentAPI.addDocumentWithFileInSite(
+				dmDocumentTitle = "DM Document Title",
+				fileName = "FileName.jpg",
+				filePath = ${filePath},
+				siteId = ${siteId});
+		}
+
+		task ("When a put request is made to update a document with an invalid file name") {
+			var documentId = JSONPathUtil.getIdValue(response = ${response});
+
+			var response = DocumentAPI.updateDocumentByDocumentId(
+				documentId = ${documentId},
+				fileName = "FileNameEdit?.jpg",
+				title = "DM Document Title Edit");
+
+			echo("####After updating the document: '${response}'");
+		}
+
+		task ("Then file name and title can not be updated") {
+			TestUtils.assertPartialEquals(
+				actual = ${response},
+				expected = "BAD_REQUEST");
+
+			TestUtils.assertPartialEquals(
+				actual = ${response},
+				expected = "Invalid file name FileNameEdit?.jpg");
+
+			User.loginPG(
+				password = "test",
+				userEmailAddress = "test@liferay.com");
+
+			DMNavigator.openToEntryInAdmin(
+				dmDocumentTitle = "DM Document Title",
+				groupName = "Guest",
+				siteURLKey = "guest");
+
+			DMDocument.viewCP(
+				dmDocumentFileName = "FileName.jpg",
+				dmDocumentTitle = "DM Document Title");
```

Comment:
Instead of checking the UI you can get send a get request to the file and assert the file name/title was the same as before. That way this test can also have webdriver be disabled since it's run completely on backend operations